### PR TITLE
[4.3][Guided Tours] Tour Edit: Enforce value in extensions

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #40026 .

### Summary of Changes
Set extensions field mandatory, so users must choose a component (or ALL)
Add a check in the store method in case of data import with empty extension field.

### Testing Instructions
see #40026

### Actual result BEFORE applying this Pull Request
Error message from the database

### Expected result AFTER applying this Pull Request
Requested input on screen, a default value if the extension field is empty. 

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_guidedtours/src/Table/TourTable.php [hunk 1]
```diff
@@ -74,6 +74,10 @@ public function store($updateNulls = true)
             $this->created = $date;
         }
 
+        if (empty($this->extensions)) {
+            $this->extensions = '["*"]';
+        }
+
```
Comment (brianteeman) [inline administrator/components/com_guidedtours/src/Table/TourTable.php]: What does this do? If the field is empty when you try to save. You just get an invalid field message. This doesn't appear to have any impact

Comment (obuisard) [reply administrator/components/com_guidedtours/src/Table/TourTable.php]: I agree, we don't need this extra piece anymore. Save is blocked if a value is missing.
You can safely remove it Christiane @chmst. Thanks!

**Comment by sdwjoomla** [comment]
I have tested this item :white_check_mark: successfully on 13729d64ca0dcb42b838e8e25718d9a5fc62ee42<br /><br />Unable to save. Received error message and missing required field indicated.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40034">issues.joomla.org/tracker/joomla-cms/40034</a>.</sub>

**Comment by obuisard** [comment]
I have tested this item :white_check_mark: successfully on 13729d64ca0dcb42b838e8e25718d9a5fc62ee42<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40034">issues.joomla.org/tracker/joomla-cms/40034</a>.</sub>

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on 13729d64ca0dcb42b838e8e25718d9a5fc62ee42<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40034">issues.joomla.org/tracker/joomla-cms/40034</a>.</sub>

**Comment by obuisard** [comment]
I have tested this item :white_check_mark: successfully on 5004c396a78e631d0b026d31925dba6f954ee760<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40034">issues.joomla.org/tracker/joomla-cms/40034</a>.</sub>

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on 5004c396a78e631d0b026d31925dba6f954ee760<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40034">issues.joomla.org/tracker/joomla-cms/40034</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40034">issues.joomla.org/tracker/joomla-cms/40034</a>.</sub>

**Comment by obuisard** [comment]
Thank you Christiane @chmst for the PR!

**Pull request merged**